unittest/client/src/ilm_control_unittests.cpp

```diff
@@ -0,0 +1,5071 @@
+ *                      and notication callback is null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the wm_listener_surface_size()
+ *                      -# Verification point:
+ *                         +# When invoke callback function, g_ilm_notification_mask must be seted to default ILM_NOTIFICATION_ALL
+ */
+TEST_F(IlmControlTest, wm_listener_surface_size_invalidSurface)
+{
+    uint32_t l_surface_id = 6;
+    wm_listener_surface_size(&ilm_context.wl, nullptr, l_surface_id, 450, 450);
+
+    ASSERT_EQ(ILM_NOTIFICATION_ALL, g_ilm_notification_mask);
+}
+
+/** ================================================================================================
+ * @test_id             wm_listener_surface_size_nullNotification
+ * @brief               Test case of wm_listener_surface_size() where valid surface id
+ *                      and notication callback is null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the wm_listener_surface_size()
+ *                      -# Verification point:
+ *                         +# When invoke callback function, g_ilm_notification_mask must be seted to default ILM_NOTIFICATION_ALL
+ */
+TEST_F(IlmControlTest, wm_listener_surface_size_nullNotification)
+{
+    uint32_t l_surface_id = 5;
+    wm_listener_surface_size(&ilm_context.wl, nullptr, l_surface_id, 0, 500);
+
+    ASSERT_EQ(ILM_NOTIFICATION_ALL, g_ilm_notification_mask);
+}
+
+/** ================================================================================================
+ * @test_id             wm_listener_surface_size_success
+ * @brief               Test case of wm_listener_surface_size() where valid surface id
+ *                      and notication callback is not null pointer
+ * @test_procedure Steps:
+ *                      -# Set the ilm context initialized is true, ready init
+ *                      -# Calling the ilm_surfaceAddNotification() to add a callback
+ *                      -# Calling the wm_listener_surface_size() multiple times with different input param width, height
+ *                      -# Verification point:
+ *                         +# ilm_surfaceAddNotification() must return ILM_SUCCESS
+ *                         +# When invoke callback function, g_ilm_notification_mask must be seted to ILM_NOTIFICATION_CONFIGURED
+ */
+TEST_F(IlmControlTest, wm_listener_surface_size_success)
+{
+    ilm_context.initialized = true;
+
+    uint32_t l_surface_id = 1;
+    ASSERT_EQ(ILM_SUCCESS, ilm_surfaceAddNotification(l_surface_id, &surfaceCallbackFunction));
+    wm_listener_surface_size(&ilm_context.wl, nullptr, l_surface_id, 0, 500);
+    wm_listener_surface_size(&ilm_context.wl, nullptr, l_surface_id, 500, 0);
+    wm_listener_surface_size(&ilm_context.wl, nullptr, l_surface_id, 500, 500);
+    wm_listener_surface_size(&ilm_context.wl, nullptr, l_surface_id, 500, 500);
+
+    ASSERT_EQ(ILM_NOTIFICATION_CONFIGURED, g_ilm_notification_mask);
```

Comment:
Here why the wm_listener_surface_size called multiple time??
Should we consider resetting g_ilm_notification_mask between calls to wm_listener_surface_size and assert for the mask set to ILM_NOTIFICATION_CONFIGURED after each call?